plugins/elantp/fu-elantp-i2c-device.c

```diff
@@ -620,13 +720,26 @@ fu_elantp_i2c_device_attach(FuDevice *device, FuProgress *progress, GError **err
 		g_prefix_error(error, "cannot enable TP report: ");
 		return FALSE;
 	}
-	if (!fu_elantp_i2c_device_write_cmd(self, 0x0306, 0x003, error)) {
-		g_prefix_error(error, "cannot switch to TP PTP mode: ");
-		return FALSE;
-	}
+
 	if (!fu_elantp_i2c_device_ensure_iap_ctrl(self, error))
 		return FALSE;
 
+	if (g_strcmp0(fu_device_get_custom_flags(device), "elantp-absolute") == 0) {
+		if (!fu_elantp_i2c_device_write_cmd(self, 0x0300, 0x001, error)) {
+			g_prefix_error(error, "cannot switch to TP ABS mode: ");
+			return FALSE;
+		}
+		if (getuid() == 0 && geteuid() == 0) {
```

Comment:
?? why would this be running as non-root?

Reply:
Because the permissions of bind and unbind are 200, and chromeOS uses a separate user/group to execute fwupd.
When updating firmware via fwupdmgr & fwupd, there will be no permission to rebind.

Reply:
@bleungatchromium is this by design? We should probably handle this better in fwupd itself.

Reply:
But so if there is no permissions to rebind will the update be able to actually run?

Reply:
> But so if there is no permissions to rebind will the update be able to actually run?

Yes, fwupd has enough permissions to access the I2C device.
The rebind action is only used to update the device information in the driver.

Reply:
IMO rather than guarding for this you should use a local error (IE `error_local`) and then catch the error type you get when failing to write due to permissions and emit a debug message then.  Something like this:

```
g_autoptr(GError) error_local = NULL;
if (!fu_elantp_i2c_device_rebind_driver(self, &error_local)) {
	if(g_error_matches(error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED))
		g_debug("%s", error_local->message);
	else {
		g_propagate_error(error, g_steal_pointer(&error_local));
		return FALSE;
	}
}
```

Switch out the error codes for the right ones.

Reply:
Many thanks, updated as suggested.